staging/src/k8s.io/client-go/tools/events/event_broadcaster.go

```diff
@@ -35,6 +35,9 @@ import (
 	"k8s.io/apimachinery/pkg/util/strategicpatch"
 	"k8s.io/apimachinery/pkg/util/wait"
 	typedv1beta1 "k8s.io/client-go/kubernetes/typed/events/v1beta1"
+	typedv1core "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/tools/record"
```

Comment:
This dependency is unfortunate, but I still think it's better to have that hidden in the library instead of requiring every single user to think about this fallback.
[The fact that we already depend on record/util below makes it even less of an issue imho.]

So I think it's fine, but just cc-ing @yastij and @liggitt - in case the will disagree.

Reply:
I'm fine with this dependency too, the benefit outweigh the cost

Reply:
dependencies are at the module level, so depending on `k8s.io/client-go` is the boundary I'm most concerned about